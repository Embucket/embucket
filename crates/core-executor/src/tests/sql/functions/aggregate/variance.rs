// Aggregate function tests for VARIANCE and VARIANCE_SAMP

use crate::test_query;

test_query!(
    variance_aggregate,
    "SELECT VARIANCE(n) AS v FROM (VALUES (1), (2), (3), (4)) AS t(n)",
    snapshot_path = "aggregate"
);

test_query!(
    variance_samp_aggregate,
    "SELECT VARIANCE_SAMP(n) AS v FROM (VALUES (1), (2), (3), (4)) AS t(n)",
    snapshot_path = "aggregate"
);

test_query!(
    variance_window,
    "SELECT VARIANCE(n) OVER () AS v FROM (VALUES (1), (2), (3), (4)) AS t(n)",
    snapshot_path = "aggregate"
);

test_query!(
    variance_samp_window,
    "SELECT VARIANCE_SAMP(n) OVER () AS v FROM (VALUES (1), (2), (3), (4)) AS t(n)",
    snapshot_path = "aggregate"
);
