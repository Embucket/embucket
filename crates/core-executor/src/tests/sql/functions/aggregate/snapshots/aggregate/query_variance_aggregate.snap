---
source: crates/core-executor/src/tests/sql/functions/aggregate/variance.rs
description: "\"SELECT VARIANCE(n) AS v FROM (VALUES (1), (2), (3), (4)) AS t(n)\""
---
Ok(
    [
        "+--------------------+",
        "| v                  |",
        "+--------------------+",
        "| 1.6666666666666667 |",
        "+--------------------+",
    ],
)
